protocol-units/bridge/chains/ethereum/src/client.rs

```diff
@@ -342,7 +401,9 @@ impl BridgeContractCounterparty for EthClient {
 			FixedBytes(hash_lock.0),
 			U256::from(time_lock.0),
 			recipient.0 .0,
-			U256::from(amount.0),
+			U256::try_from(amount.0).map_err(|e| {   
+				BridgeContractInitiatorError::ConversionError  
+			}).expect("Failed to convert amount to U256"),
```

Comment:
If the `expect` is executed, the service will panic causing a crash. https://doc.rust-lang.org/std/result/enum.Result.html#method.expect

Some of these expect calls were left in when initially writing the service, let's remove them now. Instead you could to this, where you have implemented the trait `TryFrom` set`type Error = BridgeContractCounterpartyError`

Because `lock_bridge_transfer_assets` returns `BridgeContractCounterpartyResult<()>` and this is 
`pub type BridgeContractCounterpartyResult<T> = Result<T, BridgeContractCounterpartyError>;`

You should then be able to do `U256::try_from(amount.0)?`